libgnucash/engine/gnc-datetime.cpp

```diff
@@ -339,6 +342,34 @@ GncDateTimeImpl::GncDateTimeImpl(const GncDateImpl& date, DayPart part) :
 /* Member function definitions for GncDateTimeImpl.
  */
 
+static bool
+parse_chars_into_num (const char* ptr, const char *end_ptr, char next_char, int32_t& rv)
+{
+    auto result = std::from_chars (ptr, end_ptr, rv);
+    return (result.ec == std::errc() && result.ptr == end_ptr && *result.ptr == next_char);
+}
+
+static std::optional<PTime>
+fast_iso8601_utc_parse (const char* str)
+{
+    int32_t year, month, mday, hour, min, sec;
+
+    if (str &&
+        parse_chars_into_num (str,      str +  4, '-', year) &&
+        parse_chars_into_num (str +  5, str +  7, '-', month) &&
+        parse_chars_into_num (str +  8, str + 10, ' ', mday) &&
+        parse_chars_into_num (str + 11, str + 13, ':', hour) &&
+        parse_chars_into_num (str + 14, str + 16, ':', min) &&
+        parse_chars_into_num (str + 17, str + 19, ' ', sec) &&
+        !strcmp (str + 20, "+0000"))
```

Comment:
There are two ISO-ish formats, one with and one without delimiters. I don't remember offhand which SQL backend uses which format, but we should optimize both.